web-app/django/VIM/apps/instruments/views/instrument_detail.py

```diff
@@ -14,10 +13,16 @@ class InstrumentDetail(DetailView):
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
 
-        # Query the instrument names in all languages
-        context["instrument_names"] = (
-            context["instrument"].instrumentname_set.all().select_related("language")
-        )
+        search_query = self.request.GET.get('search', '')
+        sort_by = self.request.GET.get('sort', 'language__en_label')
+
+        instrument_names = context["instrument"].instrumentname_set.all().select_related("language")
+
+        if search_query:
+            instrument_names = instrument_names.filter(language__en_label__icontains=search_query)
+
+        instrument_names = instrument_names.order_by(sort_by)
+        context["instrument_names"] = instrument_names

```

Comment:
We have a solr set-up for UMIL... let's use that for search! 

Lemme know if you want some examples for how to set this up.

Reply:
Yes, some example would be great!

Reply:
Ok, a few things would be helpful:

- The solr admin panel is location at port 8983 (so in development at localhost:8983). One of the more useful areas of this for debugging search applications is the `Query` section. In the drop-down on the left, you'll see the `virtual-instrument-museum` core. Select that one, and then in the panel that appears you can go to `Query` to get a UI for making solr queries. Useful for debugging queries. 
- An example from Cantus Ultimus: we use this view to query solr to provide search suggestions: https://github.com/DDMAL/cantus/blob/main/app/public/cantusdata/views/suggestion.py. The view is pretty simple: building a request to the solr server and then making it with the requests package.
- I don't have an example of this, but if you wanted to perform the search without routing through a Django view, you could just make the request via Javascript to the solr server (eg. have the browser request to an endpoint that your route to the solr server using the nginx configurations and then unpack the json response in the client).